common/network-common/src/test/java/org/apache/spark/network/TransportClientFactorySuite.java

```diff
@@ -217,4 +217,16 @@ public String get(String name) {
       assertFalse(c1.isActive());
     }
   }
+
+  @Test
+  public void closeFactoryBeforeCreateClient() {
+    TransportClientFactory factory = context.createClientFactory();
+    factory.close();
+    try {
+      // NullPointerException occurred if factory.createClient() after factory.close()
+      factory.createClient(TestUtils.getLocalHost(), server1.getPort());
+    } catch (Exception e) {
```

Comment:
Why not just catch NullPointerException and ignore it?
But, maybe createClient should throw a better exception to begin with?

Reply:
The test shows exception is occurred if TransportClientFactory.createClient() called after TransportClientFactory.close().
Agree to throw a better exception and the patch is updated. There should be an IOException now.